### PR TITLE
docs: update and edit syllabus to match intent for course

### PR DESCRIPTION
**Pull request by lwjohnst86**
Good first draft of the syllabus.

### CODE REVIEW

File: preamble/syllabus.qmd [hunk 1]
```diff
@@ -4,152 +4,84 @@
...
+-   Writes code that others often use or at least would like (or hopes)
+    others will use.
+-   Has a research position that often ends up having more coding or
+    programming related tasks, and would like to simplify or become more
+    efficient in writing code so that less time is spent on it.
+-   Has some familiarity with, but not much experience in, using version
+    control systems like [Git](https://git-scm.com/) as well as hosting
+    services like [GitHub](https://github.com/).
+-   Wants to get into doing more research software
+    development/engineering.
```
Comment (signekb) [inline preamble/syllabus.qmd:73]:
```suggestion
    development/engineering.
-   Wants to increase the reproducibility and usability of the code they produce
```

File: preamble/syllabus.qmd [hunk 2]
```diff
@@ -4,152 +4,84 @@
...
+-   Uses [R](https://cran.r-project.org/) for doing statistical
+    analysis, general data analysis, data processing, data science, or
+    for other research purposes.
+-   Has no or limited formal training in programming or software
+    development.
+-   Has taken some formal R courses (like one of our previous
+    [ones](https://r-cubed.rostools.org)) and feels fairly competent in
+    writing R code.
+-   Writes a lot of R scripts and some functions, often with a fair
+    amount of repetitive code used throughout.
```
Comment (signekb) [inline preamble/syllabus.qmd:63]: Do we assume that learners *often* repeat code? It doesn't really fit with the assumptions further down that they use function-workflows (L81-86) ?

**Review by signekb — COMMENTED** [review]
Nice! I have some minor comments and suggestions here and there 👍

File: preamble/syllabus.qmd (hunk 2 above)
Comment (lwjohnst86) [reply preamble/syllabus.qmd:63]: Using a function-based workflow isn't a binary state, there is lots of repetition you can still do as you progress to getting better.

**Pull request merged**